### PR TITLE
[Fix] Display QuantumModel

### LINKED ISSUE #175: Visualization for `QuantumModel` instances
Allow for calling something like:

```
from qadence.draw import display
from qadence import QuantumModel

qm = QuantumModel(...)
display(qm)
```

### PR DESCRIPTION
**Pull request by nmheim**
closes #175

### CODE REVIEW

File: qadence/draw/utils.py [hunk 1]
```diff
@@ -143,13 +143,11 @@ def _(circuit: QuantumCircuit, *args: Any, **kwargs: Any) -> QuantumCircuitDiagr
 
 @make_diagram.register
 def _(model: QuantumModel, *args: Any, **kwargs: Any) -> QuantumCircuitDiagram:
-    raise ValueError("Not yet supported.")
-
     # FIXME: include measurement icon
-    if model.out_features > 1:
+    if model.out_features > 1:  # type: ignore
```
Comment (Roland-djee) [inline qadence/draw/utils.py:147]: Did you get `mypy` warnings ?

**Review by Roland-djee — APPROVED** [review]
LGTM.

**Review by madagra — APPROVED** [review]
Thanks, @nmheim, it looks good to me. I do not really get what was the issue with the `QuantumModel` drawing before, though.

File: qadence/draw/utils.py (hunk 1 above)
Comment (nmheim) [reply qadence/draw/utils.py:147]: yes, good point actually, I'll catch the `model.out_features = None` case.

**Comment by nmheim** [comment]
> Thanks, @nmheim, it looks good to me. I do not really get what was the issue with the `QuantumModel` drawing before, though.

I just forgot to remove [this line](https://github.com/pasqal-io/qadence/pull/177/files#diff-ad3474894e1c7eeeaf1b4601e8effc2c56cd1fbbe5982fde68cbdcb225b24d08L146). We will probably get some weird wire ordering bugs for some observables, but we will fix that with #42

**Pull request merged**